pkg/bindings/containers/attach.go

```diff
@@ -29,7 +29,10 @@ type CloseWriter interface {
 	CloseWrite() error
 }
 
-// Attach attaches to a running container
+// Attach attaches to a running container.
+//
+// NOTE: When the container is configured to use a tty and stdin is a tty,
+// this currently leaks a goroutine reading from stdin even if the ctx is cancelled.
```

Comment:
I don't think this is quite correct. The go routine is leaked until the stdin is closed.